torch/csrc/jit/codegen/cuda/maxinfo_propagator.h

```diff
@@ -0,0 +1,239 @@
+// generates the maximum spanning tree that perserves the most amount of root
+// domain information from the reference tensor.
+//*
+// During the path-finding, we explicitly keep track of the information about
+// which reference tensor's root ID's information is preserved, and to which
+// level. This information is stored as a vector of `RootIDInfo`, where each
+// item in the vector corresponds to one ID in the reference tensor's root
+// domain.
+class TORCH_CUDA_CU_API MaxRootDomainInfoSpanningTree
+    : public MaxInfoSpanningTree {
+ protected:
+  // This is a struct storing how the information about a root ID in the
+  // starting tensor is preserved during path-finding. If during path-finding,
+  // we reached a tensor called the "current" tensor, we are interested in the
+  // following information:
+  // - Which reference tensor's root ID's information does the current tensor
+  //   contains? Each RootIDInfo object should correspond to one reference
+  //   tensor's root ID, but we don't need to store this ID explicitly.
+  // - For this reference tensor's root ID, what are its corresponding IDs in
+  //   the current tensor's root/rfactor domain?
+  // - Is the current tensor's information about this reference tensor's root ID
+  //   complete?
+  struct RootIDInfo {
+    // Each object of this class correspond to one root ID in the reference
+    // tensor, but we do not need to explicitly store this ID.
+
+    // The IDs in the current tensor's root or rfactor domain that contains
+    // information of the corresponding reference tensor's root ID. Whether we
+    // are using root domain or rfactor domain depends on how we reached the
+    // current tensor during path-finding. `is_rfactor` tells us whether the IDs
+    // contained in `mapped_ids` are from the root domain or the rfactor domain.
+    std::unordered_set<IterDomain*> mapped_ids;
+
+    // Does `mapped_ids` contain all the IDs required to recompute the
+    // corresponding reference tensor's root ID? For example, if we have
+    //   t1 = input tensor of shape (20,)
+    //   t2 = view(t1, {4, 5})
+    //   t3 = sum(t2, {1})
+    //   t4 = set(t3)
+    // and we start the path-finding from t1, then t2 and t3's information about
+    // t1 is complete, but t4 is not because one axis is missing.
+    bool is_complete;
+
+    // Is `mapped_ids` from the root domain or rfactor domain of the current
+    // tensor? We only store IDs from one of them, depending on how we reach the
+    // current tensor during path-finding. If we reached the current tensor from
+    // a consumer, then `mapped_ids` containes IDs in the current tensor's
+    // rfactor domain because the rfactor domain contains raw information. If we
+    // reached the current tensor from a producer, then `mapped_ids` containes
+    // IDs in the current tensor's root domain because the root domain contains
+    // raw information.
+    bool is_rfactor;
+
+    RootIDInfo() = default;
```

Comment:
Can you delete this? It's causing build failures internally.

Not quite sure what the difference in compilers is... but if you replace `= default;` with `{}` then I think you repro the build failure

Reply:
Out of curiosity, which compiler/version are you guys using in your internal build?

Reply:
> FYI, apparently this fails in C++20 which is why it was failing internally

Ha it makes sense. 
Aggregate initialization changed at c++20 and we can no longer have `user-declared` constructors. https://en.cppreference.com/w/cpp/language/aggregate_initialization